### PR TITLE
[WIP][SYSTEMML-1216] implement local svd( ) function

### PR DESCRIPTION
**Pull request by j143**
Please merge this PR for the local `svd()` function. Implementing distributed `svd` may take some time ( I will open that in a different PR), by then my branch may become stale. Thanks. 

@niketanpansare @mboehm7

### CODE REVIEW

**Comment by j143** [comment]
@mboehm7 - first, we only add this function for convenience. As per your suggestion, We can experiment with both the `svd()` with and and w/o local `svd()` (??).

**Comment by j143** [comment]
Test this Jenkins

**Comment by j143** [comment]
Test this Jenkins

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1864/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1868/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1872/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1873/

**Comment by mboehm7** [comment]
Well, as expressed before, adding more of such special case operators concerns me a bit. However, I think we should go ahead (and resolve this consistently for all these operations). Could you please add a test case in org.apache.sysml.test.integration.functions.unary.matrix though (you could follow the template of LUFactorizeTest or QRFactorizeTest).

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>6</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1876/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseMR</strong></a></li></ul>

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>6</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1882/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseMR</strong></a></li></ul>

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/<h2>Failed Tests: <span class='status-failure'>7</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>7</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.data/FullReblockTest/testTextCellMultipleMSparseMR/'><strong>org.apache.sysml.test.integration.functions.data.FullReblockTest.testTextCellMultipleMSparseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1900/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseMR</strong></a></li></ul>

**Comment by j143** [comment]
I believe the `unitary`, `orthogonality` and `reproduction` of matrix `A`, will be sufficient to verify the operation. However, I have some I/O problems with the scripts. The only output matrix is `D`.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>6</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1927/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseMR</strong></a></li></ul>

**Comment by j143** [comment]
Somebody help me with this. Thanks.  At `B = U %*% S %*% t(V)`
```
ERROR: ./src/test/scripts/functions/unary/matrix/svd.dml line 51, column 6 -- 
Hops::optimizeMMChain() : Matrix Dimension Mismatch: 1 != 2500
```

**Comment by mboehm7** [comment]
Sorry, I somehow missed this - I'll take a look tonight.

File: src/test/scripts/functions/unary/matrix/svd.dml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+I1 = U %*% t(U);
+I2 = V %*% t(V);
+
+Urows = nrow(U);
+Ucols = ncol(U);
+Vrows = nrow(V);
+Vcols = ncol(V);
+
+# check for, whether the U and V are square or not.
+if ( Urows == Ucols & Vrows == Vcols) {
```
Comment (mboehm7) [inline src/test/scripts/functions/unary/matrix/svd.dml]: if you change the test to non-square matrix, please also change this condition to ensure a and b are computed.

Comment (j143) [inline src/test/scripts/functions/unary/matrix/svd.dml]: As `U` & `V` are always square according to [wikipedia](https://en.wikipedia.org/wiki/Singular_value_decomposition#Statement_of_the_theorem), let us test for square (??).

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1950/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1951/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1952/

**Comment by j143** [comment]
@deroneriksson can the CI test be delayed by 2 or 3 minutes, instead of instant trigger. Thanks.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1953/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/<h2>Failed Tests: <span class='status-failure'>8</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>8</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseSP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseSP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1955/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseSP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseSP</strong></a></li></ul>

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/<h2>Failed Tests: <span class='status-failure'>8</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>8</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseSP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseSP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1961/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseSP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseSP</strong></a></li></ul>

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1966/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/<h2>Failed Tests: <span class='status-failure'>8</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>8</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testSVDFactorizeDenseSP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testSVDFactorizeDenseSP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseHybrid/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseHybrid</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseCP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseCP</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseMR/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseMR</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1969/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.functions.unary.matrix/SVDFactorizeTest/testLargeSVDFactorizeDenseSP/'><strong>org.apache.sysml.test.integration.functions.unary.matrix.SVDFactorizeTest.testLargeSVDFactorizeDenseSP</strong></a></li></ul>

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1970/

**Comment by j143** [comment]
@mboehm7 - sir, this have not passed the squareness test. Other than that we are good to go.

**Comment by mboehm7** [comment]
thanks for addressing the review comments @j143. This LGTM.

**Pull request closed without merging**